### PR TITLE
Isaac/arm timeout 2023.2

### PR DESCRIPTION
**Pull request by IsaacCalligeros95**
Adds the default proxy to calls using the AzureResourceManagementClient.
Includes fix for release/2023.2 that is included in the latest tag, but not the release branch for S3 feeds downloading on targets. 

## Before
image (1)

## After
image

### CODE REVIEW

File: source/Calamari.AzureResourceGroup/DeployAzureResourceGroupBehaviour.cs [hunk 1]
```diff
@@ -75,7 +75,7 @@ public async Task Execute(RunningDeployment deployment)
             Func<Task<IResourceManagementClient>> createArmClient = async () =>
                                                               {
                                                                   var token = new TokenCredentials(await ServicePrincipal.GetAuthorizationToken(tenantId, clientId, password, resourceManagementEndpoint, activeDirectoryEndPoint));
-                                                                  var resourcesClient = new ResourceManagementClient(token)
+                                                                  var resourcesClient = new ResourceManagementClient(token, AuthHttpClientFactory.ProxyClientHandler())
```
Comment (benPearce1) [inline source/Calamari.AzureResourceGroup/DeployAzureResourceGroupBehaviour.cs:78]: What happens here if there is no proxy configured?
Does `WebRequest.DefaultWebProxy` being set to no proxy mean that it just works directly?

File: source/Calamari.Shared/Integration/Packages/Download/S3PackageDownloader.cs [hunk 2]
```diff
@@ -64,6 +64,7 @@ string BuildFileName(string prefix, string version, string extension)
             }
 
             var cacheDirectory = PackageDownloaderUtils.GetPackageRoot(feedId);
+            fileSystem.EnsureDirectoryExists(cacheDirectory);
```
Comment (benPearce1) [inline source/Calamari.Shared/Integration/Packages/Download/S3PackageDownloader.cs:67]: do the changes in this file belong somewhere else?

File: source/Calamari.AzureResourceGroup/DeployAzureResourceGroupBehaviour.cs (hunk 1 above)
Comment (IsaacCalligeros95) [reply source/Calamari.AzureResourceGroup/DeployAzureResourceGroupBehaviour.cs:78]: Without a proxy configured, the ResourceManagementClient just makes calls regardless of the default proxy

Comment (IsaacCalligeros95) [reply source/Calamari.AzureResourceGroup/DeployAzureResourceGroupBehaviour.cs:78]: The images in the PR description show this

**Review by hnrkndrssn — APPROVED** [review]
lgtm 👍

**Pull request merged**